src/patterns/security-code/index.md.njk

```diff
@@ -0,0 +1,135 @@
+
+Asking users to enter a security code sent to their mobile phone gives a second layer of security over a [password](/patterns/passwords/).
+
+You can ask for a security code every time a user signs in or only once per device, depending upon the risk of your service.
+
+## How it works
+
+Send and ask the user for the security code when they: 
+* create an account
+* sign in later
+
+### When creating an account
+
+When the user creates an account, ask for their password and mobile phone number at the same time. Make it clear if you will only use the mobile number for sending security codes, or if you will use it for other purposes.
+
+After saving the user’s password and mobile phone number, verify their mobile phone number by sending them a text message with a 5 digit code in this format:
+
+{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
+{{ govukInsetText({
+  text: "12345 is your [service name] security code"
+}) }}
+
+Then ask the user to enter this code:
+
+{{ example({group: "patterns", item: "security-code", example: "default", html: true, nunjucks: true, open: false}) }}
+
+Ignore any formatting or punctuation such as spaces, hyphens and dashes.
+
+Give a time limit of 15 minutes for users to enter the code. If this time limit expires, display a ‘code has expired’ error message and send the user a new code.
+
+If the user follows the ‘Not received a text message?’ link, allow them to check which mobile number they entered, and to change it if necessary. This prevents the user becoming stuck if they entered a mobile number incorrectly. Do not allow users to change the number when signing in.
+
+{{ example({group: "patterns", item: "security-code", example: "resend-first-time", html: true, nunjucks: true, open: false}) }}
+
+This should send a new code, with its own 15 minute expiry window. The previous code should remain valid within its own expiry time.
+
+### When signing in
+
+When a user returns to your service, verify their password first. Once they have entered this correctly, send a text message to their mobile phone with a 5 digit code in this format:
+
+{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
+{{ govukInsetText({
+  text: "12345 is your [service name] security code"
+}) }}
+
+Ask the user to enter this code. Use the same pattern and time limit as when creating an account.
+
+If they follow the ‘Not received a text message?’ link, show them a page allowing them to resend the code. Do not reveal the mobile number you sent it to.
+
+{{ example({group: "patterns", item: "security-code", example: "resend", html: true, nunjucks: true, open: false}) }}
+
+You should tell users what to do if they no longer have access to the phone used to sign up.
+
+Depending upon your service, they may need to contact a support address, or may be able to ask a colleague to update their phone number.
```

Comment:
May -> might, minor content style thing:
```suggestion
Depending upon your service, they might need to contact a support address, or ask a colleague to update their phone number.
```
Also, in the last part of the sentence it's bit confusing who's colleague is being referred to here? The user's or the support agent's? Not sure if this solution is common enough to be worth mentioning.

Reply:
@calvin-lau-sig7  I wrote this - the scenario imagined is where the user is using a service professionally as part of an organisation, and are set up within a team within that service, where any member of a team (or perhaps only admins) can reset the password or phone number of a colleague within that team – instead of needing to contact support. I think perhaps GOV.UK Notify has this model?

This might be uncommon enough or too confusing to mention though? Could be dropped.

Reply:
To be honest, I'm feeling like it might be more trouble than it's worth to explain. 
I could try to write the idea of another 'seat' on the account, etc. but it feels like it's going into another pattern.

So simplify it to:
```suggestion
For example, you might tell them to contact a support address.
```